src/discovery/mod.rs

```diff
@@ -0,0 +1,172 @@
+use crate::{
+    schema::{
+        self,
+        ws_discovery::{probe, probe_matches},
+    },
+    soap,
+};
+use async_std::stream::StreamExt;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+#[derive(Debug)]
+pub enum Error {
+    Network(std::io::Error),
+    Internal(String),
+}
+
+pub async fn discover(
```

Comment:
can you add comments here on how to use this function?

Reply:
Sure. I've also changed its signature to make it more useful with `StreamExt` combinators.